### PR TITLE
Added TAB shortcut to focus the chat textbox when it is not in focus.…

### PR DESCRIPTION
**Pull request by ftwpala**
… Added setting for max messages (extracted from command) and added a disable option for chat combos.

Demo of TAB focus https://i.imgur.com/w2e8vfD.gif

### CODE REVIEW

File: assets/chat/js/chat.js [hunk 1]
```diff
@@ -861,7 +874,7 @@ class Chat {
         if (
             win.lastmessage &&
             win.lastmessage.type === MessageTypes.EMOTE &&
-            win.lastmessage.emotecount > 1
+            win.lastmessage.emotecount > 1
```
Comment (jbpratt) [inline assets/chat/js/chat.js:877]:
```suggestion
            win.lastmessage.emotecount > 1
```

File: assets/chat/js/menus.js [hunk 2]
```diff
@@ -237,6 +240,34 @@ class ChatSettingsMenu extends ChatMenu {
...
+            return;
+        }
+        let errorMessage = $("#maximum-messages-error");
+
+        // Reset error message
+        errorMessage.removeAttr("style");
+
+        let data = $(e.target).val();
+
+        if(data ==""){
```
Comment (jbpratt) [inline assets/chat/js/menus.js:256]:
```suggestion
        if(data == ""){
```

File: assets/chat/js/menus.js [hunk 3]
```diff
@@ -237,6 +240,34 @@ class ChatSettingsMenu extends ChatMenu {
...
+        } 
+
+        const newmaxlines = Math.abs(parseInt(data, 10));
+        if (!newmaxlines || (newmaxlines < 25 && newmaxlines > 500)) {
+            $("#maximum-messages-error").css("display", "inline-block");
+        }
+        else{
+            this.chat.settings.set("maxlines", data);
+            this.chat.applySettings();
+            this.chat.commitSettings();
```
Comment (jbpratt) [inline assets/chat/js/menus.js:267]:
```suggestion
            this.chat.commitSettings();
```

**Review by jbpratt — CHANGES_REQUESTED** [review]
my being nit-picky :smiley_cat:

**Review by SoMuchForSubtlety — APPROVED** [review]

**Review by slugalisk — APPROVED** [review]

**Pull request merged**